mirage/views/osf-resource.ts

```diff
@@ -49,61 +52,63 @@ export function osfResource(
     }
 
     if (actions.includes('show')) {
-        server.get(detailPath, modelName);
+        server.get(detailPath, mirageModelName);
     }
 
     if (actions.includes('create')) {
-        server.post(opts.path, modelName);
+        server.post(opts.path, mirageModelName);
     }
 
     if (actions.includes('update')) {
-        server.patch(detailPath, modelName);
-        server.put(detailPath, modelName);
+        server.patch(detailPath, mirageModelName);
+        server.put(detailPath, mirageModelName);
     }
 
     if (actions.includes('delete')) {
-        server.del(detailPath, modelName);
+        server.del(detailPath, mirageModelName);
     }
 }
 
-export function osfNestedResource(
+export function osfNestedResource<K extends keyof ModelRegistry>(
     server: Server,
-    parentModelName: string,
-    relationshipName: string,
+    parentModelName: K,
+    relationshipName: string & RelationshipsFor<ModelRegistry[K]>,
     options?: Partial<NestedResourceOptions>,
 ) {
     const opts: NestedResourceOptions = Object.assign({
         allow: ['list'],
-        path: `/${underscore(parentModelName)}/:parentID/${underscore(relationshipName)}`,
+        path: `/${pluralize(underscore(parentModelName))}/:parentID/${pluralize(underscore(relationshipName))}`,
```

Comment:
I'm not sure pluralizing the relationshipName is always the right thing here. It seems that the relationship names are already pluralized, and, if one wasn't, it would be intentional.

Reply:
Good catch, I was caught up pluralizing everythings 👍